library/common/http/header_utility.h

```diff
@@ -17,12 +17,20 @@ namespace Utility {
 std::string convertToString(envoy_data s);
 
 /**
- * Transform envoy_headers to HeaderMap.
+ * Transform envoy_headers to RequestHeaderMap.
+ * This function copies the content.
+ * @param headers, the envoy_headers to transform.
+ * @return HeaderMapPtr, the HeaderMap 1:1 transformation of the headers param.
+ */
+RequestHeaderMapPtr toRequestHeaders(envoy_headers headers);
+
+/**
+ * Transform envoy_headers to RequestHeaderMap.
  * This function copies the content.
  * @param headers, the envoy_headers to transform.
  * @return HeaderMapPtr, the HeaderMap 1:1 transformation of the headers param.
  */
-HeaderMapPtr toInternalHeaders(envoy_headers headers);
+RequestTrailerMapPtr toRequestTrailers(envoy_headers headers);
```

Comment:
nit: might also make sense to rename `headers` to `trailers` now that we have an explicit `RequestTrailerMapPtr` type

Reply:
Thoughts?